### PR TITLE
Bug fix rational

### PR DESCRIPTION
**Pull request by nishnik**
Denominator can not be zero. Right?

### CODE REVIEW

**Comment by nishnik** [comment]
Sorry.
It included the previous commits too.

File: symengine/rational.cpp [hunk 1]
```diff
@@ -110,7 +110,7 @@ bool Rational::is_perfect_power(bool is_expected) const
 
     const mpz_class &den = i.get_den();
     if (den == 0)
-        return mpz_perfect_power_p(num.get_mpz_t()) != 0;
+        throw std::runtime_error("Rational: Division by zero.");
```
Comment (nishnik) [inline symengine/rational.cpp]: Before creating a PR, I wanted to be sure.
Is this right? @certik @thilinarmtb

Comment (isuruf) [inline symengine/rational.cpp]: Yes, go ahead and make a new PR

**Comment by nishnik** [comment]
Superseded by #805

**Pull request closed without merging**